lucene/queries/src/java/org/apache/lucene/queries/function/ValueSourceScorer.java

```diff
@@ -94,4 +97,17 @@ public float getMaxScore(int upTo) throws IOException {
     return Float.POSITIVE_INFINITY;
   }
 
+  /**
+   * Cost evaluation function which defines the cost of access for the TwoPhaseIterator for this class
+   * This method should be overridden for specifying custom cost methods. Used by {@link TwoPhaseIterator#matchCost()}
+   * for the instance owned by this class
+   *
+   * @return cost of access
+   *
+   * @lucene.experimental
+   */
+  protected float costEvaluationFunction() {
```

Comment:
Can we call this simply matchCost now since, after all, the TPI.matchCost just calls it now with no mutable override?